backend/src/src-vulnerabilities/src/main/java/com/siemens/sw360/vulnerabilities/VulnerabilityHandler.java

```diff
@@ -80,20 +82,115 @@ public VulnerabilityHandler() throws IOException, SW360Exception {
         return dtos;
     }
 
-    private List<VulnerabilityDTO> getVulsByReleaseId(String releaseId, User user){
-        List<ReleaseVulnerabilityRelation> relations = dbHandler.getRelationsByReleaseIds(Arrays.asList(releaseId));
-        if (relations == null || relations.size() ==0){
+    @Override
+    public List<VulnerabilityDTO> getVulnerabilitiesByProjectId(String projectId, User user) throws TException {
+        if (!PermissionUtils.isUserAtLeast(UserGroup.USER, user)){
             return Collections.emptyList();
         }
-        return relations
-                .stream()
-                .map(rel -> rel.getVulnerabilityId())
-                .map(id -> dbHandler.getById(Vulnerability.class, id))
-                .map(VulnerabilityMapper::createVulnerabilityDTO)
-                .map(dto -> enrichVulnerabilityDTO(dto, releaseId, user))
+        Set<String> releaseIds = projectDatabaseHandler.getProjectById(projectId, user).getReleaseIdToUsage().keySet();
+        if (releaseIds == null || releaseIds.size() == 0){
+            return Collections.emptyList();
+        }
+
+        List<VulnerabilityDTO> dtos = new ArrayList<>();
+        releaseIds.stream()
+                .map(id -> getVulsByReleaseId(id, user))
+                .forEach(list -> dtos.addAll(list));
+
+        return dtos;
+    }
+
+    @Override
+    public List<VulnerabilityDTO> getVulnerabilitiesByReleaseIdWithoutIncorrect(String releaseId, User user) throws TException {
+        if (!PermissionUtils.isUserAtLeast(UserGroup.USER, user)){
+            return Collections.emptyList();
+        }
+
+        return getVulsByReleaseIdWithoutIncorrect(releaseId, user);
+    }
+
+    @Override
+    public List<VulnerabilityDTO> getVulnerabilitiesByComponentIdWithoutIncorrect(String componentId, User user) throws TException {
+        if (!PermissionUtils.isUserAtLeast(UserGroup.USER, user)){
+            return Collections.emptyList();
+        }
+        List<Release> releases = compHandler.getReleasesFromComponentId(componentId, user);
+        if (releases == null || releases.size() == 0){
+            return Collections.emptyList();
+        }
+
+        List<VulnerabilityDTO> dtos = new ArrayList<>();
+        releases.stream()
+                .map(rel -> getVulsByReleaseIdWithoutIncorrect(rel.getId(), user))
+                .forEach(list -> dtos.addAll(list));
```

Comment:
same as above. also, these two snippets are so much alike... is it worth to extract a method?